cluster/service/cluster_service_blackbox_test.go

```diff
@@ -630,6 +631,54 @@ func (s *ClusterServiceTestSuite) TestListForAuth() {
 	})
 }
 
+func (s *ClusterServiceTestSuite) TestDelete() {
+
+	s.T().Run("ok", func(t *testing.T) {
+		// given
+		c := test.CreateCluster(t, s.DB)
+		sa := &authtestsupport.Identity{
+			Username: auth.ToolChainOperator,
+			ID:       uuid.NewV4(),
+		}
+		ctx, err := authtestsupport.EmbedServiceAccountTokenInContext(context.Background(), sa)
+		require.NoError(t, err)
+		// when
+		err = s.Application.ClusterService().Delete(ctx, c.ClusterID)
+		// then
+		require.NoError(t, err)
+		// check that
+		c, err = repository.NewClusterRepository(s.DB).Load(ctx, c.ClusterID)
+		require.Error(t, err)
+		assert.IsType(t, errors.NotFoundError{}, err)
+		require.Nil(t, c)
+	})
+
+	s.T().Run("failures", func(t *testing.T) {
+		t.Run("unauthorized", func(t *testing.T) {
+			// given
+			c := test.CreateCluster(t, s.DB)
+
+			for _, saName := range []string{"fabric8-auth", "fabric8-oso-proxy", "fabric8-tenant", "fabric8-jenkins-idler", "fabric8-jenkins-proxy", "other"} {
+				t.Run(saName, func(t *testing.T) {
+					// given
+					sa := &authtestsupport.Identity{
+						Username: saName,
+						ID:       uuid.NewV4(),
+					}
+					ctx, err := authtestsupport.EmbedServiceAccountTokenInContext(context.Background(), sa)
+					require.NoError(t, err)
+					// when
+					err = s.Application.ClusterService().Delete(ctx, c.ClusterID)
+					// then
+					require.Error(t, err)
```

Comment:
require.Error is not needed here. testsupport.AssertError() will do it for you.

Reply:
ok, Removed in 870041f